tests/test_fio_workload.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+import logging
+
+from ocs import constants, exceptions
+from ocs import workload
+from ocsci.testlib import ManageTest
+from utility.utils import TimeoutSampler
+from utility import templating
+
+
+from tests.fixtures import (
+    create_rbd_storageclass, create_pod, create_pvc, create_ceph_block_pool,
+    create_rbd_secret
+)
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.usefixtures(
+    create_rbd_secret.__name__,
+    create_ceph_block_pool.__name__,
+    create_rbd_storageclass.__name__,
+    create_pvc.__name__,
+    create_pod.__name__
+)
+class TestFIOWorkload(ManageTest):
+
+    def test_fio_with_block_storage(self):
+        name = 'test_workload'
+        spec = self.pod_obj.data.get('spec')
+        path = spec['containers'][0]['volumeMounts'][0]['mountPath']
+        work_load = 'fio'
+        storage_type = 'fs'
+        # few io parameters for Fio
+        runtime = 10
+        size = '200M'
+
+        wl = workload.WorkLoad(
+            name, path, work_load, storage_type, self.pod_obj
+        )
+        assert wl.setup()
+        io_params = templating.load_yaml_to_dict(constants.FIO_IO_PARAMS_YAML)
+        io_params['runtime'] = runtime
+        io_params['size'] = size
+
+        future_result = wl.run(**io_params)
```

Comment:
I went through mongodb workload, the procedure for mongodb is like this
1. Two set of yamls are involved in this, 
     a. one for setting up mongodb service (this has around 4 yamls like deployment, service etc)
     b. Another for workload YCSB (benchmarking tool) 

In case if user wants to change any mongodb config we can fetch mongodb yaml config as usually we do, also ycsb config. This is still in setup() phase where deployment happens, so this setup config dictionary will be forwarded to corresponding workload's setup() which will be a function in workload/mongodb/mongod.py. 

After this we run phase, where we will have run_params dictionary for workload, so calling <mondb.py>.run() will run the command on ycsb pod.
I feel current infra is enough to do this thing. Also same for postgresql as well.

Please let me know in case you feel something needs to be changed.

Reply:
Thanks for clarifying - looks good